### ISSUE TITLE
NRT BNF Mapping

### ISSUE THREAD

**Issue by brianmackenna** [ebmdatalab/openprescribing#1695]
I was having a quick look at nicotine replacement products over the last 9 years (see [Jupyter notebook](https://github.com/ebmdatalab/jupyter-notebooks/blob/master/nonCCG%20Random/Nicotine.ipynb))

Buprenorphine is returned in the results. In 2011 0410020D0AAACAC = Buprenorphine_Tab Subling 400mcg S/F, whereas today the 0410020D0AAACAC = 
![image](https://user-images.githubusercontent.com/40460354/57756989-61ff6e00-76ec-11e9-9ddb-f9cd413bcd83.png)

Just wanted to check if we had missed it when mapping or if it is a BSA issue that will need fixing there?

**Comment by sebbacon** [comment]
```sql

SELECT
  month,
  bnf_name,
  bnf_code
FROM
  ebmdatalab.hscic.normalised_prescribing_standard
WHERE
  bnf_code = '0410020D0AAACAC'
GROUP BY
  month,
  bnf_name,
  bnf_code
ORDER BY
  month
```

![image](https://user-images.githubusercontent.com/211271/57761098-b7d81400-76f4-11e9-9a8d-741bf7d9112a.png)

**Comment by sebbacon** [comment]
```sql
SELECT
  *
FROM
  ebmdatalab.hscic.bnf_map
WHERE
  former_bnf_code = '0410020D0AAACAC'
  OR current_bnf_code = '0410020D0AAACAC'
```

![image](https://user-images.githubusercontent.com/211271/57761207-f5d53800-76f4-11e9-9eb9-8483ad4b8249.png)

**Comment by sebbacon** [comment]
To summarise:

* `0410020D0AAACAC` is currently `Nicotine Bitartrate_Loz 2mg S/F (F/Mint)`, whose BNF code used to be `0410000D0AAACAC`; that code is no longer in the BNF
* `Buprenorphine_Tab Subling 400mcg S/F` is listed twice in our analyse form under `0407020B0AAADAD` and `0410030A0AAACAC`
* The nicotine presentation is the correct one

**Comment by sebbacon** [comment]
```sql
SELECT
  month,
  bnf_code,
  bnf_name
FROM
  [hscic.prescribing]
WHERE
  bnf_code = '0410000D0AAACAC'
  OR bnf_code = '0410020D0AAACAC'
GROUP BY
  month,
  bnf_code,
  bnf_name
ORDER BY month
```
![image](https://user-images.githubusercontent.com/211271/57762725-ed323100-76f7-11e9-8e02-75e224d45f96.png)

**Comment by sebbacon** [comment]
To summarise:
* According to the raw data, the code `0410000D0AAACAC` (associated with the Buprenorphine) disappeared in Sept 2011; and the code `0410020D0AAACAC` (associated with the Nicotine) appeared in Jan 2012. 
* According to our mapping data, `0410000D0AAACAC` is now `0410020D0AAACAC`
* `0410020D0AAACAC` is currently `Nicotine Bitartrate_Loz 2mg S/F (F/Mint)`, whose BNF code used to be `0410000D0AAACAC`; that code is no longer in the BNF
* `Buprenorphine_Tab Subling 400mcg S/F` is listed twice in our analyse form under `0407020B0AAADAD` and `0410030A0AAACAC` - because it's in two different chapters (analgesics, and substance dependence)
* The substance dependence one was last seen in the data as 0410000D0AAACAC in October 2011, and as 0410030A0AAACAC from November.

**Comment by inglesp** [comment]
To summarise the summary:

* There are two BNF codes of interest:
  * `0410000D0AAACAC` (associatied with the Buprenorphine)
  * `0410020D0AAACAC` (associated with the Nicotine Bitartrate)
* According to our mapping data, `0410000D0AAACAC` (Buprenorphine) is now `0410020D0AAACAC` (Nicotine Bitartrate)
* In the raw prescribing data, `0410000D0AAACAC` (Buprenorphine) disappeared in Oct 2011, and `0410020D0AAACAC` (Nicotine Bitartrate) appeared in Jan 2012.

The main question is how did our mapping get created?

**Comment by inglesp** [comment]
> The main question is how did our mapping get created?

(this turns out not to be the main question)

`presentation_replacements/2012.txt` includes the following (several times):

```
0410000	0410020
```

This was added in a4ca955a.

This is saying that a presentation whose BNF code is `0410020ABCDEFGH` replaces a presentation with BNF code `0410000ABCDEFGH`.  This will be reflected in the `bnf_mapping` table in BQ, which in turn means that prescribing for `0410000ABCDEFGH` will appear, in `normalised_prescribing_standard` as being for `0410020ABCDEFGH`.

(The `generate_presentation_replacements` task finds all presentations beginning with  `0410020`.  For each such presentation P1, it takes the BNF code, replaces the section code with `0410000` to create another BNF code, finds or creates a presentation P2 with this other BNF code, and marks P2 as being replaced by P1.  The [logic](https://github.com/ebmdatalab/openprescribing/blob/master/openprescribing/frontend/management/commands/generate_presentation_replacements.py#L153-L165) is a bit hard to follow but I'm confident it's correct.)

It's not clear to me why the extract from `presentation_replacements/2012.txt` contains duplicates.  I don't think it matters here.

**Comment by inglesp** [comment]
In our `frontend_bnf` table:

```
# select bnf_code, name, replaced_by_id from frontend_presentation where bnf_code = '0410000D0AAACAC';
┌─────────────────┬──────────────────────────────────────────┬─────────────────┐
│    bnf_code     │                   name                   │ replaced_by_id  │
├─────────────────┼──────────────────────────────────────────┼─────────────────┤
│ 0410000D0AAACAC │ Nicotine Bitartrate_Loz 2mg S/F (F/Mint) │ 0410020D0AAACAC │
└─────────────────┴──────────────────────────────────────────┴─────────────────┘
```

This is something with the old code and the new name.  I don't know where this came from.  It is not in the oldest code list from the [ISP](https://apps.nhsbsa.nhs.uk/infosystems/data/showDataSelector.do?reportId=126) (2015).

Also in our `frontend_bnf` table:

```
# select bnf_code, name, replaced_by_id from frontend_presentation where name = 'Buprenorphine_Tab Subling 400mcg S/F';
┌─────────────────┬──────────────────────────────────────┬─────────────────┐
│    bnf_code     │                 name                 │ replaced_by_id  │
├─────────────────┼──────────────────────────────────────┼─────────────────┤
│ 0407020B0AAADAD │ Buprenorphine_Tab Subling 400mcg S/F │ ¤               │
│ 0410000A0AAACAC │ Buprenorphine_Tab Subling 400mcg S/F │ 0410030A0AAACAC │
│ 0410030A0AAACAC │ Buprenorphine_Tab Subling 400mcg S/F │ ¤               │
└─────────────────┴──────────────────────────────────────┴─────────────────┘
```

This shows that there is an Buprenorphine presentation in with a similar BNF code in `041003`.  Could there have been a mistake in the mapping that confused the two?

**Comment by inglesp** [comment]
The real main question:

In the prescribing data, was a prescription with BNF code `0410000D0AAACAC` for Buprenorphine or for Nicotine Bitartrate?

There are four options:

* It was for Nicotine Bitartrate, and the code in the prescribing data was right (but the name was wrong)
* It was for Buprenorphine, and the code in the prescribing data was wrong (but the name was right)
* It was for Buprenorphine with the correct code, which was in the wrong part of the BNF, and we haven't been told that it has moved
* It was for Buprenorphine with the correct code, which was in the wrong part of the BNF, and we have been told that it has moved, but we missed this

Without knowing how the raw prescribing data was generated, I don't think we have enough information to pick an option.
